scripts/build/Jenkins/Jenkinsfile

```diff
@@ -336,51 +416,281 @@ def PreBuildCommonSteps(Map pipelineConfig, String snapshot, String repositoryNa
+            ext = '.sh'
+        }
         else command += '.cmd'
         command += " -u ${pipelineConfig.BUILD_ENTRY_POINT} --platform ${platform} --type ${type}"
-        dir("${workspace}/${ENGINE_REPOSITORY_NAME}") {
-            palSh(command, "Running ${platform} ${type}")
+
+        dir(workspace) {
+            def extrasAutomatedTestingControlFile = {}
+            dir(EXTRAS_REPOSITORY_NAME) {
+                echo 'Loading Extras .automatedtesting.json'
+                extrasAutomatedTestingControlFile = readJSON file: '.automatedtesting.json'
+            }
+            dir(ENGINE_REPOSITORY_NAME) {
+                //register the engine
+                PlatformSh("scripts/o3de${ext} register --this-engine", "Registering this engine, will generate the o3de_manifest")
+
+                //register all the objects specified in this repos .automatedtesting.json
+                extrasAutomatedTestingControlFile.REGISTER_ENGINES.each { engineName ->
+                    PlatformSh("scripts/o3de${ext} register --engine-path ${workspace}/${EXTRAS_REPOSITORY_NAME}/${engineName} || ver>nul", "Registering engine in ${workspace}/${EXTRAS_REPOSITORY_NAME}/${engineName}")
+                }
+                extrasAutomatedTestingControlFile.REGISTER_PROJECTS.each { projectName ->
+                    PlatformSh("scripts/o3de${ext} register --project-path ${workspace}/${EXTRAS_REPOSITORY_NAME}/${projectName} || ver>nul", "Registering project in ${workspace}/${EXTRAS_REPOSITORY_NAME}/${projectName}")
+                }
+                extrasAutomatedTestingControlFile.REGISTER_GEMS.each { gemName ->
+                    PlatformSh("scripts/o3de${ext} register --gem-path ${workspace}/${EXTRAS_REPOSITORY_NAME}/${gemName} || ver>nul", "Registering gem in ${workspace}/${EXTRAS_REPOSITORY_NAME}/${gemName}")
+                }
+                extrasAutomatedTestingControlFile.REGISTER_TEMPLATES.each { templateName ->
+                    PlatformSh("scripts/o3de${ext} register --template-path ${workspace}/${EXTRAS_REPOSITORY_NAME}/${templateName} || ver>nul", "Registering template in ${workspace}/${EXTRAS_REPOSITORY_NAME}/${templateName}")
+                }
+                extrasAutomatedTestingControlFile.REGISTER_RESTRICTED.each { restrictedName ->
+                    PlatformSh("scripts/o3de${ext} register --restricted-path ${workspace}/${EXTRAS_REPOSITORY_NAME}/${restrictedName} || ver>nul", "Registering restricted in ${workspace}/${EXTRAS_REPOSITORY_NAME}/${restrictedName}")
+                }
+                //enable all gems specified in the control file
+                extrasAutomatedTestingControlFile.ENABLE_GEMS.each { gemName ->
+                    PlatformSh("scripts/o3de${ext} enable-gem --gem-path ${workspace}/${EXTRAS_REPOSITORY_NAME}/${gemName} --project-name AutomatedTesting || ver>nul", "Enabling ${workspace}/${EXTRAS_REPOSITORY_NAME}/${gemName} for AutomatedTesting")
+                }
+            }
```

Comment:
Does the o3de_manifest.json get destroyed when the workspace is unmounted?

Otherwise there could be issues with one running a build in another workspace using a different extra or external repo causing another avenue for build failures due to attempting to use incorrect repo configuration

i.e A user registers a new gem with the extra repo and that change builds in AR, but has not been checked in yet).
Then another user builds their non-extra repo change in AR, and because the build machine node happens to be the same it tries to use the repo data from another branch.

Reply:
Does the o3de_manifest.json get destroyed when the workspace is unmounted? Yes.

Reply:
Ok that is ideal.
It isolates the builds from each other and prevents cross branch issues.